desktop/core/src/desktop/api2.py

```diff
@@ -688,6 +704,137 @@ def share_document(request):
     'document': doc.to_dict()
   })
 
+@api_error_handler
+@require_POST
+def handle_submit(request):
+  # Extract the task name and params from the request
+  try:
+    data = json.loads(request.body)
+    task_name = data.get('taskName')
+    task_params = data.get('taskParams')
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
+    keep_days = task_params.get('keep-days')
+    task_kwargs = {
+      'keep_days': keep_days,
+      'user_id': request.user.id,
+      'username': request.user.username,
+    }
+    # Enqueue the document cleanup task with keyword arguments
+    task = document_cleanup_task.apply_async(kwargs=task_kwargs)
+
+    # Return a response indicating the task has been scheduled
+    return JsonResponse({
+      'taskName': task_name,
+      'taskParams': task_params,
+      'status': 'Scheduled',
+      'task_id': task.id,  # The task ID generated by Celery
+    })
+
+  elif task_name == 'tmp clean up':
+    cleanup_threshold = task_params.get('threshold for clean up')
+    disk_check_interval = task_params.get('disk check interval')
+    task_kwargs = {
+      'username': request.user.username,
+      'cleanup_threshold': cleanup_threshold,
+      'disk_check_interval': disk_check_interval
+    }
+    task = check_disk_usage_and_clean_task.apply_async(kwargs=task_kwargs)
+
+    return JsonResponse({
+      'taskName': task_name,
+      'status': 'Scheduled',
+      'task_id': task.id,  # The task ID generated by Celery
+    })
+
+  return JsonResponse({
+    'status': 0
+  })
+
+redis_client = redis.Redis(host='localhost', port=6379, db=0)
```

Comment:
This doesn't feel right outside the methods and the older implementation looked comparatively better for the time being.

We should ideally aim for a minimal Hue-Redis client, think of scenarios outside of taskserver where we could benefit with Redis like caching configs or saving some flags or preserving state for diff Gunicorn processes.

Reply:
Going back to the old implementation for the time being. I'll note it down, Lets discuss the advantages and how to proceed further for next iteration.